awx_collection/plugins/modules/tower_workflow_job_template_schema.py

```diff
@@ -0,0 +1,378 @@
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: tower_workflow_job_template_node
+author: "John Westcott IV (@john-westcott-iv), Sean Sullivan (@Wilk42)"
+version_added: "2.9"
+short_description: create, update, or destroy Ansible Tower workflow job template graphs.
+description:
+    - Create, update, or destroy Ansible Tower workflow job template graphs.
+    - Use this to build a graph for a workflow, which dictates what the workflow runs.
+    - Replaces the deprecated tower_workflow_template module schema command.
+    - This module takes a josn list of inputs to first create nodes, then link them, and does not worry about ordering.
+      For failsafe referencing of a node, specify identifier, WFJT, and organization.
+      With those specified, you can choose to modify or not modify any other parameter.
+options:
+    workflow_job_template:
+      description:
+        - The workflow job template the node exists in.
+        - Used for looking up the node, cannot be modified after creation.
+      required: True
+      type: str
+      aliases:
+        - workflow
+    organization:
+      description:
+        - The organization of the workflow job template the node exists in.
+        - Used for looking up the workflow, not a direct model field.
+      type: str
+    schema:
+      description:
+        - A json list of nodes and their coresponding options.
+      type: str
+    destroy_current_schema:
+      description:
+        - Set in order to destroy current schema on the workflow.
+        - This option is used for full schema update, if not used, nodes not described in schema will persist and keep current associations and links.
+      type: Bool
+      default: False
+schema_options:
+    extra_data:
+      description:
+        - Variables to apply at launch time.
+        - Will only be accepted if job template prompts for vars or has a survey asking for those vars.
+      type: dict
+      default: {}
+    inventory:
+      description:
+        - Inventory applied as a prompt, if job template prompts for inventory
+      type: str
+    scm_branch:
+      description:
+        - SCM branch applied as a prompt, if job template prompts for SCM branch
+      type: str
+    job_type:
+      description:
+        - Job type applied as a prompt, if job template prompts for job type
+      type: str
+      choices:
+        - 'run'
+        - 'check'
+    job_tags:
+      description:
+        - Job tags applied as a prompt, if job template prompts for job tags
+      type: str
+    skip_tags:
+      description:
+        - Tags to skip, applied as a prompt, if job tempalte prompts for job tags
+      type: str
+    limit:
+      description:
+        - Limit to act on, applied as a prompt, if job template prompts for limit
+      type: str
```

Comment:
I'm confused about these options, you said the `schema` is the important input and takes a list of nodes. These are node properties, so they shouldn't be inputs to the module, they should be nested properties in dicts inside the schema option, if I read it correctly.

Reply:
@AlanCoding I'm planning in the future to do a simplified version along with the tower_export schema model, but it should be updated now.